solidity/contracts/bank/Bank.sol

```diff
@@ -254,6 +257,31 @@ contract Bank is Ownable {
         _increaseBalance(recipient, amount);
     }
 
+    /// @notice Increases the given smart contract `vault`'s balance and
+    ///         notifies the `vault` contract. Called by the Bridge after
+    ///         the deposits routed by depositors to that `vault` have been
+    ///         swept by the Bridge. This way, the depositor does not have to
+    ///         issue a separate  transaction to the `vault` contract.
+    ///         Can be called only by the Bridge.
+    /// @dev The `vault` must implement `IVault` intrface.
+    /// @param vault Address of `IVault` recipient contract
+    /// @param depositors Addresses of depositors whose deposits have been swept
+    /// @param depositedAmounts Amounts deposited by individual depositors and
+    ///        swept. The `vault`'s balance in the Bank will be increased by the
+    ///        sum of all elements in this array.
+    function increaseBalanceAndCall(
+        address vault,
+        address[] calldata depositors,
+        uint256[] calldata depositedAmounts
```

Comment:
Shouldn't we require both arrays to have an equal length and `depositors` length is >0? Also, if we do it here we could make it clear in the contract between `Bank` and `IVault` interface that `onBalanceIncreased` is always called with valid parameters. This way we only pay once for the validation and implementations don't need to bother with that. If we decide to do that, it would be good to update the `@dev` doc.

Reply:
Yes, let's do it. We have length validation in `increaseBalances`, we should also have it here.

Reply:
Done in https://github.com/keep-network/tbtc-v2/pull/94/commits/a4370e66f09b8da92eb7f14eb941b4c329a6baba.